### ISSUE TITLE
[prefer-function-type] Ignore interfaces with JSDoc comment

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#3488]
**Repro**

```TS
interface Int {
  /**
   * JSDoc comment.
   */
  (): string;
}
```

**Expected Result**

I suggest an option to leave the above code as is.

**Actual Result**

It is converted to:

```TS
/**
 * JSDoc comment.
 */
type Int =(): string;
```

**Additional Info**

There is a reason to keep interface when there is JSDoc.

At least in Visual Studio Code, if you have interface the editor will show function documentation on mouse hover.

If you convert interface to type, the editor stops showing documentation.

So, I would even add an option to prefer interface over type when there is JSDoc comment.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.0` |
| `@typescript-eslint/parser`        | `4.26.0` |
| `TypeScript`                       | `4.3.2` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.17.0` |

**Comment by bradzacher** [comment]
Sounds more like a feature request for TypeScript.
I don't think that we should be scanning for this case.

**Comment by ilyub** [comment]
https://github.com/microsoft/TypeScript/issues/44469
